alexandria/core/views.py

```diff
@@ -100,6 +108,44 @@ def update(self, request, *args, **kwargs):
 
         return response
 
+    @action(methods=["get"], detail=True)
+    def convert(self, request, pk=None):
+        if not settings.ALEXANDRIA_ENABLE_PDF_CONVERSION:
+            raise ValidationError(_("PDF conversion is not enabled."))
+
+        document = self.get_object()
+        file = document.get_latest_original()
+
+        response = requests.post(
+            settings.ALEXANDRIA_DMS_URL + "/convert",
+            data={"target_format": "pdf"},
+            headers={"authorization": get_authorization_header(request)},
+            files={"file": file.content},
+        )
+
+        if response.status_code == HTTP_401_UNAUTHORIZED:
+            raise AuthenticationFailed(_("Token has expired."))
+
+        response.raise_for_status()
+
+        converted_document = models.Document.objects.create(
+            title={k: v + ".pdf" for k, v in document.title.items()},
+            description=document.description,
+            category=document.category,
+            date=document.date,
+        )
+        file_name = file.name + ".pdf"
```

Comment:
Does this mean that `foo.docx` will become `foo.docx.pdf`?

Also, same question goes for the document title.

Reply:
Yes, not sure if trying to clean up the file extensions makes sense. As a file name can be anything

Reply:
using [splitext](https://docs.python.org/3/library/os.path.html#os.path.splitext) for cleaning